### PR TITLE
Re-enabling ClientWebSocket close and cancel tests in UWP

### PR DESCRIPTION
**Pull request by Diego-Perez-Botero**
Re-enabling ClientWebSocket's CloseTest and CancelTest runs for UWP.

All of these tests pass after a minor behavioral fix, where the "cancel pending ReceiveAsync" scenario was resulting in a different exception type compared to .NET Framework and non-UWP .NET Core. The following CancelTest test cases caught the bad behavior that I'm fixing as part of this PR:

- ReceiveAsync_CancelThenReceive_ThrowsOperationCanceledException
- ReceiveAsync_ReceiveThenCancel_ThrowsOperationCanceledException
- ReceiveAsync_AfterCancellationDoReceiveAsync_ThrowsWebSocketException

Contributes to #20132

### CODE REVIEW

File: src/System.Net.WebSockets.Client/tests/CloseTest.cs [hunk 1]
```diff
@@ -257,6 +256,7 @@ public class CloseTest : ClientWebSocketTestBase
             }
         }
 
+        [ActiveIssue(20362, TargetFrameworkMonikers.Uap)]
```
Comment (Diego-Perez-Botero) [inline src/System.Net.WebSockets.Client/tests/CloseTest.cs]: I've updated #20362 explaining that the UWP ClientWebSocket implementation mirrors the .NET Framework behavior, leading to this test failing. Isn't the .NET Framework behavior the "gold standard", though?

I'm wondering whether this test method should be updated to pin the two types of behaviors instead of having an active issue. It seems to me like .NET Framework's behavior isn't buggy; it's just different from non-UWP .NET Core. Thoughts?

File: src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs [hunk 2]
```diff
@@ -547,21 +554,6 @@ private void UpdateState(WebSocketState value)
                 _state = value;
             }
         }
-
-        private void UpdateStateCloseReceived()
```
Comment (Diego-Perez-Botero) [inline src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs]: Removing unused method

**Comment by Diego-Perez-Botero** [comment]
@dotnet-bot Test Outerloop Windows x86 Release Build
@dotnet-bot Test Outerloop Linux x64 Release Build
@dotnet-bot Test Outerloop UWP CoreCLR x64 Debug Build

**Comment by Diego-Perez-Botero** [comment]
**Outerloop Windows x86 Release Build** and **Outerloop Linux x64 Release Build** test failures are due to known ManagedHandler_HttpClientHandler_SslProtocols_Test issues (#21923 #21924 #21925 #21927 #21917).

**Comment by davidsh** [comment]
>Outerloop Windows x86 Release Build and Outerloop Linux x64 Release Build test failures are due to known ManagedHandler_HttpClientHandler_SslProtocols_Test issues (#21923 #21924 #21925 #21927 #21917).

The tests should be disabled when the managed handler is running.  We shouldn't keep getting CI failures in Outerloop.

cc: @stephentoub 

The SslProtocols Test need code like this to skip if managedhandler is being injected:

```c#
if (ManagedHandlerTestHelpers.IsEnabled)
{
    return;
}
```

**Comment by davidsh** [comment]
These failures:
>System.Net.Http.Functional.Tests.ManagedHandler_HttpClientHandler_SslProtocols_Test

are now affecting my current PR.  I will disable these tests on managed handler and push a new commit in #22075

**Comment by stephentoub** [comment]
> due to known ManagedHandler_HttpClientHandler_SslProtocols_Test issues

The issues were closed because they were believed to be fixed.

**Comment by Diego-Perez-Botero** [comment]
@dotnet-bot Test Outerloop Linux x64 Release Build
@dotnet-bot Test Outerloop Windows x86 Release Build
@dotnet-bot test Outerloop UWP NETNative x86 Release Build

File: src/System.Net.WebSockets.Client/tests/CloseTest.cs (hunk 1 above)
Comment (Diego-Perez-Botero) [inline src/System.Net.WebSockets.Client/tests/CloseTest.cs]: Regardless, #20362 will be addressed as a separate PR.

**Comment by Diego-Perez-Botero** [comment]
Still waiting for this PR to get some traction.
